lib/HeadersGeneration/ModelToHeaderPipe.cpp

```diff
@@ -24,42 +23,65 @@ static llvm::cl::opt<bool> InlineTypes("inline-types",
 
 namespace revng::pipes {
 
+inline constexpr char ModelHeaderFileContainerMIMEType[] = "text/x.c+ptml";
+inline constexpr char ModelHeaderFileContainerSuffix[] = ".h";
+inline constexpr char ModelHeaderFileContainerName[] = "model-header";
+using ModelHeaderFileContainer = FileContainer<&kinds::ModelHeader,
+                                               ModelHeaderFileContainerName,
+                                               ModelHeaderFileContainerMIMEType,
+                                               ModelHeaderFileContainerSuffix>;
+
+class ModelToHeader {
+public:
+  static constexpr auto Name = "model-to-header";
+
+  std::array<pipeline::ContractGroup, 1> getContract() const {
+    using namespace pipeline;
+    using namespace revng::kinds;
+
+    Contract C1(Binary, 0, ModelHeader, 1, InputPreservation::Preserve);
+    return { ContractGroup({ C1 }) };
+  }
+
+  // TODO: BinaryFile here is a placeholder. In principle this pipe has no real
+  // input container. It just juses the model in Ctx to generated HeaderFile.
+  // At the moment revng-pipeline does not support pipes with no inputs, so we
+  // had to resort to this trick. Whenever pipes with no inputs are supported
+  // BinaryFile can be dropped.
+  void run(const pipeline::ExecutionContext &Ctx,
+           const BinaryFileContainer &BinaryFile,
+           ModelHeaderFileContainer &HeaderFile) {
+
+    std::error_code EC;
+    llvm::raw_fd_ostream Header(HeaderFile.getOrCreatePath(), EC);
+    if (EC)
+      revng_abort(EC.message().c_str());
+
+    ptml::HeaderBuilder
+      B = ptml::CTypeBuilder(Header, { .EnableTypeInlining = InlineTypes });
+    B.printModelHeader(*getModelFromContext(Ctx));
+
+    Header.flush();
+    EC = Header.error();
+    if (EC)
+      revng_abort(EC.message().c_str());
+  }
+
+  llvm::Error checkPrecondition(const pipeline::Context &Ctx) const {
+    return llvm::Error::success();
+  }
+
+  void print(const pipeline::Context &Ctx,
+             llvm::raw_ostream &OS,
+             llvm::ArrayRef<std::string> Names) const {
+    OS << *revng::ResourceFinder.findFile("bin/revng");
+    OS << " model to-header -yaml -i=model.yml -o=" << Names[1] << "\n";
+  }
+};
```

Comment:
Make the body of this method empty. It's already informally deprecated and will be removed soon.

Reply:
ale's branch went in already, so this is not relevant anymore :tada: